test/functional/interface_zmq.py

```
@@ -173,13 +179,20 @@ def setup_zmq_test(self, services, *, recv_timeout=60, sync_blocks=True, ipv6=Fa
+    def test_basic(self, unix = False):
+        self.log.info(f"Running basic test with {'ipc' if unix else 'tcp'} protocol")
 
         # Invalid zmq arguments don't take down the node, see #17185.
         self.restart_node(0, ["-zmqpubrawtx=foo", "-zmqpubhashtx=bar"])
         self.zmq_context = zmq.Context()
 
         address = f"tcp://127.0.0.1:{self.zmq_port_base}"
+
+        if unix:
+            # Use the shortest temp path possible since paths may have as little as 92-char limit
+            socket_path = tempfile.NamedTemporaryFile().name
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)